runtime/lib/capi/ExecutionContext.hpp

```diff
@@ -72,11 +110,16 @@ class SharedLibraryManager final {
         // If you have compiled this file with sanitizers and you reach this line
         // you will get an error.
         // Please re-compile without sanitizers.
+
 #ifdef __APPLE__
         // macOS doesn't support RTLD_DEEPBIND
         _handler = dlopen(filename.c_str(), RTLD_LAZY);
 #else
-        _handler = dlopen(filename.c_str(), RTLD_LAZY | RTLD_DEEPBIND);
+        // Closing the dynamic library of Lightning simulators with dlclose() where
+        // OpenMP directives are in use would raise memory segfaults
+        // Ali: Adding RTLD_NODELETE fixed the issue.
+
+        _handler = dlopen(filename.c_str(), RTLD_LAZY | RTLD_NODELETE);
 #endif
```

Comment:
Similar comment to flag below.

Reply:
The code is clear in this method and is not repeated anywhere else in the runtime. Do you still want this suggestion?

Reply:
I do have a preference to avoid preprocessor macros anywhere. But I understand that it is just a preference. The example I showed, I think, is a lot easier to read than having two functionalities (potentially 3, when later we have to support windows). I'll leave this as your choice whether you disagree or not.